decidim-admin/app/commands/decidim/admin/create_participatory_process_admin.rb

```diff
@@ -33,21 +34,39 @@ def call
 
       private
 
-      attr_reader :form, :participatory_process
+      attr_reader :form, :participatory_process, :current_user
+
+      def create_role
+        role = :admin if form.roles.include?("process_admin")
+        role = :collaborator if form.roles.include?("collaborator")
 
-      def create_participatory_process_admin
         ParticipatoryProcessUserRole.create!(
-          role: :admin,
+          role: role,
           user: user,
           participatory_process: @participatory_process
         )
       end
 
       def user
-        User.where(
+        @user ||= User.where(
           email: form.email,
           organization: participatory_process.organization
-        ).first
+        ).first || InviteUser.call(user_form) do
+          on(:ok) do |user|
+            return user
+          end
+        end
+      end
+
+      def user_form
+        OpenStruct.new({
+          name: form.name,
+          email: form.email.downcase,
+          organization: participatory_process.organization,
+          roles: form.roles,
+          invited_by: current_user,
+          invitation_instructions: "invite_admin"
```

Comment:
We should review this, I think the invitation instructions say something specific about admins